main.py

```diff
@@ -12,6 +12,95 @@
 
 # Import whatever libraries/modules you need
 
-import numpy as np
+import matplotlib.pyplot as plt
```

Comment:
Please break your code up into logical functions that each handle one piece of logic. I will allow you to decide what each function does.